### PR TITLE
actionify change policy API

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
*Issue #, if available:*
#292 

*Description of changes:*
actionify add policy API, refactor our API code so we have an action name to each API

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by qreshi — DISMISSED** [review]

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ChangePolicy.kt [hunk 1]
```diff
@@ -51,12 +54,33 @@ data class ChangePolicy(
...
+        out.writeCollection(include)
+    }
+
     companion object {
         const val POLICY_ID_FIELD = "policy_id"
         const val STATE_FIELD = "state"
         const val INCLUDE_FIELD = "include"
         const val IS_SAFE_FIELD = "is_safe"
 
+        fun fromStreamInput(sin: StreamInput): ChangePolicy {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ChangePolicy.kt:70]: Can you just use a constructor instead?

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/StateFilter.kt [hunk 2]
```diff
@@ -41,5 +49,11 @@ data class StateFilter(val state: String) {
 
             return StateFilter(requireNotNull(state) { "Must include a state when using include filter" })
         }
+
+        fun fromStreamInput(sin: StreamInput): StateFilter {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/StateFilter.kt:53]: Same as above, can we just use constructor?

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/TransportChangePolicyAction.kt [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        private val managedIndexUuids = mutableListOf<Pair<String, String>>()
+        private val indexUuidToCurrentState = mutableMapOf<String, String>()
+        lateinit var policy: Policy
+        lateinit var response: GetResponse
+        private var updated: Int = 0
+
+        fun start() {
+            val getRequest = GetRequest(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX, request.changePolicy.policyID)
+
+            client.get(getRequest, object : ActionListener<GetResponse> {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/TransportChangePolicyAction.kt:89]: Did this not work? `client.get(getRequest, ActionListener.wrap(::onGetPolicyResponse, ::onFailure))`

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/ChangePolicyRequest.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        changePolicy: ChangePolicy
+    ) : super() {
+        this.indices = indices
+        this.changePolicy = changePolicy
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        indices = sin.readStringList(),
+        changePolicy = sin.readOptionalWriteable { ChangePolicy.fromStreamInput(it) } as ChangePolicy
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/ChangePolicyRequest.kt:42]: Is this optional? What does. request look like without a change policy?
Assuming it's not optional just change this to `changePolicy = ChangePolicy(sin)` now that you have a constructor.

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ChangePolicy.kt [hunk 5]
```diff
@@ -51,12 +62,24 @@ data class ChangePolicy(
...
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/ChangePolicy.kt:79]: Remove this fn, you shouldn't need it

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/ChangePolicyRequest.kt (hunk 4 above)
Comment (bowenlan-amzn) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/ChangePolicyRequest.kt]: only have writeOptionalWriteable and readOptionalWriteable, so we need to use them,

**Review by dbbaughe — APPROVED** [review]

**Review by qreshi — APPROVED** [review]

**Pull request merged**